### ISSUE TITLE
Add unit support in multiply and other spectrum operations

### ISSUE THREAD

**Issue by erwanp** [radis/radis#109]
### Problem

As mentioned in https://github.com/radis/radis/pull/101#pullrequestreview-394154396, the problem with the normalization of #101  is that [multiply](https://github.com/radis/radis/blob/develop/radis/spectrum/operations.py#L480) does not use dimensioned units. 

> multiply does not currently handle multiplications with dimensioned coefficients. These should change the units of the output Spectrum.

### Suggested fix

(this is related to #57 , which focused on Spectrum methods, while here we have operations between 2 Spectra, or 1 Spectrum or arrays/floats) 

- [x] use astropy.units for the conversion. We now have (#106 ) a radis.phys.units.Unit based on astropy.units:
```
a = b * c 
unit_a = (Unit(unit_b) * Unit(unit_c)).to_string() 
```

- [x] add tests!
Such as: 
```
Radiance(s) -= 0.01 * Unit("W/cm2/sr/µm")
```

- [x] same for all the others: add, divide, etc.

Finally, some cleaning:

- [x] Implement a new normalize() method with the different options of get_residual() (that uses our new dimensioned multiply method)
- [ ] Replace most of the normalizing code of get_residual() with a simple call to `get_diff(s1.normalize(*args), s2.normalize(*args) )` ? 
- [ ] that should eventually fix #102. Check and close Issue.

- [ ] Also i think we can do the same in .plot() , if we know have a normalize method()

Final remark:

how do you normalize a transmittance spectrum? It does not really make sense, but is not forbidden here. Should raise a NotImplementedError or ValueError.

**Comment by minouHub** [comment]
I'll begin with the easy stuff. 

- If it is not physical to normalize a transmittance, the user should be informed. A `ValueError` with the following warning should do the work "A normalized Transmittance spectrum has no physical meaning".

- @erwanp. You gave clear instruction, thanks. Can you also close all the other PR and Issues related please ? I guess you're the only one who has a broad view on this problem, and for me it would help to have one channel to follow.

**Comment by erwanp** [comment]
Did, I closed #101 . 

As for your first answer:

- In general, I think that normalizing shouldn't be possible if there are multiple spectral arrays. Let's say there are absorbance (A) and absorption coefficient (k). All are tightly bound to one another : A = k * L ... you can't normalize A and k at the same time. 

Note that this was already a problem in some functions, and fixed by using [_get_unique_var](https://github.com/radis/radis/blob/develop/radis/spectrum/operations.py#L450) which we should use here too.

**Comment by erwanp** [comment]
Hello, we should get this one fixed before 0.9.26 gets released next week.
